creator-node/src/routes/files.js

```diff
@@ -506,11 +555,14 @@ const getDirCID = async (req, res) => {
 
   const cacheKey = getStoragePathQueryCacheKey(path)
 
-  let storagePath = await redisClient.get(cacheKey)
+  let storagePath = await redisClient.get(cacheKey) // can be overridden later by computed storage location if cache doesn't exist
+  let CID = null // this gets assigned later after the correct storagePath is set
+  let queryResults
+
   if (!storagePath) {
-    // Don't serve if not found in DB.
+    // We need to lookup the CID so we can correlate with the filename (eg original.jpg, 150x150.jpg)
     // Query for the file based on the dirCID and filename
-    const queryResults = await models.File.findOne({
+    queryResults = await models.File.findOne({
```

Comment:
the structure currently seems to be:

check path from redis
if not found, fetch file entry from DB and compute expected path
stream from expectedStoragePath
if that fails, then check if expectedStoragePath diff from DB storagePath
if so, stream from DB storagePath
if that also fails, fetch file from network and save to disk
stream from newly saved storagepath
the below seems more correct i think?

check path from redis
if found, attempt to stream
if stream fails or redis path not found, query file record from DB
attempt to stream file from DB storagePath
if stream fails, compute expected storagePath and attempt to stream from there (idk if this part is necessary?)
if that also fails, fetch file from network and save to disk
stream from newly saved storagePath
also:

should we update the redis cache key when we find the correct storagepath, otherwise it'll be wrong for every subsequent stream and never updated?

Reply:
resolved offline, pushing up changes